test/cookie/cookies.js

```diff
@@ -599,3 +599,37 @@ test('Set-Cookie parser', () => {
   headers = new Headers()
   assert.deepEqual(getSetCookies(headers), [])
 })
+
+test('Cookie setCookie throws if headers is not of type Headers', () => {
+  class Headers {
+    [Symbol.toStringTag] = 'CustomHeaders'
+  }
+  const headers = new Headers()
+  assert.throws(
+    () => {
+      setCookie(headers, {
+        name: 'key',
+        value: 'Cat',
+        httpOnly: true,
+        secure: true,
+        maxAge: 3
+      })
+    },
+    new TypeError('Illegal invocation')
+  )
+})
+
+test('Cookie setCookie does not throw if headers is an instance of a custom Headers class', () => {
+  class Headers {
+    [Symbol.toStringTag] = 'Headers'
+    append () { }
+  }
+  const headers = new Headers()
```

Comment:
```suggestion
  const headers = new globalThis.Headers()
```

Reply:
No bro, this is not the point of the brandCheck. But I will add one with the global this one. OK?

Reply:
done, good? :)

Reply:
it does now cover all cases of "Header"-classes

Should I add for all cookie functions these tests? getCookies, getSetCookie etc?

Reply:
It definitely is the point of the brand check. I implemented the non-strict mode to work with both the global headers and undici's headers as it didn't require internal state.

Reply:
Done